compiler/modules/front/expr.bal

```diff
@@ -574,6 +580,29 @@ function codeGenArithmeticBinaryExpr(ExprContext cx, bir:BasicBlock bb, bir:Arit
     return { result, block: bb };
 }
 
+function isNonPanickingIntOp(bir:Operand lhs, bir:Operand rhs, bir:ArithmeticBinaryOp op) returns boolean {
+    t:IntSubtypeConstraints? lhsConstraints = t:intSubtypeConstraints(lhs.semType);
+    t:IntSubtypeConstraints? rhsConstraints = t:intSubtypeConstraints(rhs.semType);
+    if lhsConstraints == () || rhsConstraints == () {
+        return false;
+    }
+    match op {
+        "+"|"-" => {
+            // largest type that can't overflow is unsigned32 (positive) and signed32 (negative)
+            return lhsConstraints.max <= 0xffffffff && rhsConstraints.max <= 0xffffffff &&
```

Comment:
Use predefined constants in lang.int e.g. `int:UNSIGNED32_MAX_VALUE`.

In the match expression, set two variables noPanicMin and noPanicMax. Then check them against each constraint in a loop at the end.

```
foreach var c in constraints {
    if c.min < noPanicMin || c.max > noPanicMax {
        return false
    }
}
return true;
```

Reply:
fixed